cmd/uid_provider/api/handlers/handler.go

```diff
@@ -0,0 +1,128 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"gitlab.inspr.dev/inspr/core/cmd/uid_provider/client"
+	"gitlab.inspr.dev/inspr/core/pkg/rest"
+)
+
+// Handler is a structure which cointains methods to handle
+// requests received by the UID Provider API
+type Handler struct {
+	rdb client.RedisManager
+	ctx context.Context
+}
+
+// NewHandler instantiates a new Handler structure
+func NewHandler(rdb client.RedisManager) *Handler {
+	return &Handler{
+		rdb: rdb,
+	}
+}
+
+// CreateUserHandler handles user creation requests
+func (h *Handler) CreateUserHandler(rw http.ResponseWriter, r *http.Request) {
+	type ReceivedDataCreate struct {
+		UID string
+		Pwd string
+		Usr client.User
+	}
+
+	data := ReceivedDataCreate{}
+	if err := json.NewDecoder(r.Body).Decode(&data); err != nil {
+		rest.ERROR(rw, err)
+		return
+	}
+
+	if err := h.rdb.CreateUser(h.ctx, data.UID, data.Pwd, data.Usr); err != nil {
+		rest.ERROR(rw, err)
+		return
+	}
+}
+
+// DeleteUserHandler handles user deletion requests
+func (h *Handler) DeleteUserHandler(rw http.ResponseWriter, r *http.Request) {
+	type ReceivedDataDelete struct {
+		UID            string
+		Pwd            string
+		UsrToBeDeleted string
```

Comment:
Can't this be UserToBeDeleted, I mean, the name is already long, why not use the full name?

Also, declare these structs on a different 'models' folder, so that they are accessible from other packages.